cliff.toml

```diff
@@ -0,0 +1,82 @@
+# git-cliff ~ configuration file
+# https://git-cliff.org/docs/configuration
+
+[changelog]
+# template for the changelog header
+header = """
+# Changelog\n
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).\n
+"""
+# template for the changelog body
+# https://keats.github.io/tera/docs/#introduction
+body = """
+{% if version -%}
+    ## [{{ version | trim_start_matches(pat="v") }}] - {{ timestamp | date(format="%Y-%m-%d") }}
+{% else -%}
+    ## [Unreleased]
+{% endif -%}
+{% for group, commits in commits | filter(attribute="merge_commit", value=false) | group_by(attribute="group") %}
+    ### {{ group | upper_first }}
+    {% for commit in commits %}
+        {%- set short_message = commit.message | split(pat="\n") | first | trim -%}
+        - {{ short_message | trim }}\n\
+    {% endfor %}
+{% endfor %}\n
+"""
+# template for the changelog footer
+footer = """
+{% for release in releases -%}
+    {% if release.version -%}
+        {% if release.previous.version -%}
+            [{{ release.version | trim_start_matches(pat="v") }}]: \
+                https://github.com/{{ remote.github.owner }}/{{ remote.github.repo }}\
+                    /compare/{{ release.previous.version }}..{{ release.version }}
+        {% endif -%}
+    {% else -%}
+        [unreleased]: https://github.com/{{ remote.github.owner }}/{{ remote.github.repo }}\
+            /compare/{{ release.previous.version }}..HEAD
+    {% endif -%}
+{% endfor %}
+<!-- generated by git-cliff -->
+"""
+# remove the leading and trailing whitespace from the templates
+trim = true
+
+[git]
+# parse the commits based on https://www.conventionalcommits.org
+conventional_commits = false
+# filter out the commits that are not conventional
+filter_unconventional = true
+# process each line of a commit as an individual commit
+split_commits = false
+# regex for parsing and grouping commits
+commit_parsers = [
+  { message = "^.*: add", group = "Added" },
+  { message = "^.*: support", group = "Added" },
+  { message = "^.*: remove", group = "Removed" },
+  { message = "^.*: delete", group = "Removed" },
+  { message = "^test", group = "Fixed" },
+  { message = "^fix", group = "Fixed" },
+  { message = "^.*: fix", group = "Fixed" },
+  { message = "^.*", group = "Changed" },
+]
+# protect breaking changes from being skipped due to matching a skipping commit_parser
+protect_breaking_commits = false
+# filter out the commits that are not matched by commit parsers
+filter_commits = false
+# regex for matching git tags
+tag_pattern = "v[0-9].*"
```

Comment:
Won't this patter allow for e.g. `v4hello`? We could include regex from `https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string` (and link to this site in comment as regex would be very complex).

Reply:
Or you could use `v[0-9]*\.[0-9]*\.[0-9]*(-rc\.[0-9]*)?`
Not tested but it shoud allow all `v<number>.<number>.<number>[-rc.<number>]` with rc being optional.

Reply:
We don't seem to worry about this in other repos. We even trigger CIs on `v*` tags. I don't think anyone will be pushing tags like `v4hello`

Reply:
I haven't seen `cliff.toml` in any other repo but I guess if you want to leave it in such a state then ok.
Let's hope we don't need have to fix it in the future.

> We even trigger CIs on v*

CIs there were possibly done before we had procedures about release versioning & tags.
And as this is fairly new repository without any workflow any deficiencies right now might get included when we get to creating one.

Here's a link to our versioning scheme: https://docs.zarhus.com/development-process/standard-release-process/#versioning-scheme